guide/docs/popular-topics/permissions.mdx

```diff
@@ -1,8 +1,382 @@
+as it doesn't have any permissions-related methods on it either.
+
+### In interactions
+
+When working with interactions (i.e., with application commands or components), there are two main methods
+defined on <DocsLink reference="disnake.Interaction" />: <DocsLink reference="disnake.Interaction.permissions" />
+(for getting interaction author's permissions) and <DocsLink reference="disnake.Interaction.app_permissions" />
+(for getting bot's permissions in interaction's channel).
+
+## How permissions are calculated
+
+Permissions can be a tough thing to understand at first, but understanding how permissions are
+applied can help you <!-- feel the pain of us mere mortals re-implementing discord client's obscure behaviour -->
+better grasp this concept.
+
+In case we're calculating permissions for a member, the strategy is as follows:
+
+-   If member is the guild owner, all permissions are granted - no questions asked. Otherwise..
+-   First we apply the `@everyone` role's permissions.
+-   Then, for each role the member has we apply it's permissions too.
+-   -   If at this point the member has the "Administrator" permission, all others are granted automatically
+        (the member is a guild-wide administrator, and, as such, is functionally equivalent to the owner except
+        they can't delete the guild).
+-   After that, if we're in a channel, the following is additionally done.
+-   -   We apply it's overwrites in the same order: `@everyone` overwrites,
+        role overwrites and finally member overwrites.
+-   -   Then [channel-specific implicit permissions](#channel-specific-implicit-permissions) are handled.
+-   In the end, we take into account [guild-wide implicit permissions](#guild-wide-implicit-permissions).
+
+In case of calculating permissions for a role, the strategy is the same as if we were calculating permissions
+for a member with only that role (and `@everyone`), except that member overwrites are
+not taken into account.
+
+:::note
+Curious about the implementation? You can see it [here](https://git.disnake.dev/disnake/blob/master/disnake/abc.py#L651-L808).
+:::
+
+### Implicit permissions
+
+In Discord, there are some permissions that, when revoked, basically make it impossible to utilize other
+granted permissions. For example, take a user with two permissions: "View Channel" and
+"Send Messages". If at some point user loses the "View Channel" permission, it wouldn't matter whether
+they can send messages in that channel: they just can't see the channel in the UI. disnake's permission system
+attempts to handle such cases via "implicit permissions" - permissions that, when granted or, more commonly,
+revoked, automatically grant (revoke) other permissions as well. The currently handled cases are separated
+into two parts: guild-wide rules and channel-specific rules, and are described below.
+
+#### Guild-wide implicit permissions
+
+Guild-wide implicit permissions currently consist of a single rule.
```

Comment:
this line seems unnecessary

Reply:
94f9216d037592b33d3c38f2adf516f62bb5736f